arch/cortex-m/src/lib.rs

```diff
@@ -395,44 +395,42 @@ unsafe fn kernel_hardfault_arm_v7m(faulting_stack: *mut u32) {
 #[naked]
 pub unsafe extern "C" fn hard_fault_handler_arm_v7m() {
     let faulting_stack: *mut u32;
-    let kernel_stack: bool;
+    let kernel_stack: u32;
 
     // First need to determine if this a kernel fault or a userspace fault.
-    llvm_asm!(
-    "
-    mov    r1, 0     /* r1 = 0 */
-    tst    lr, #4    /* bitwise AND link register to 0b100 */
-    itte   eq        /* if lr==4, run next two instructions, else, run 3rd instruction. */
-    mrseq  r0, msp   /* r0 = kernel stack pointer */
-    addeq  r1, 1     /* r1 = 1, kernel was executing */
-    mrsne  r0, psp   /* r0 = userland stack pointer */"
-    : "={r0}"(faulting_stack), "={r1}"(kernel_stack)
-    :
-    : "cc"
-    : "volatile" );
-
-    if kernel_stack {
+    asm!(
+        "mov    r1, 0     /* r1 = 0 */",
+        "tst    lr, #4    /* bitwise AND link register to 0b100 */",
+        "itte   eq        /* if lr==4, run next two instructions, else, run 3rd instruction. */",
+        "mrseq  r0, msp   /* r0 = kernel stack pointer */",
+        "addeq  r1, 1     /* r1 = 1, kernel was executing */",
+        "mrsne  r0, psp   /* r0 = userland stack pointer */",
+        out("r0") faulting_stack,
+        out("r1") kernel_stack,
+        options(nomem, nostack)
+    );
+
```

Comment:
As you outlined, we can't use `bool` for outputs anymore which makes sense. Maybe it'd be helpful to insert either a comment on the `kernel_stack` variable stating that this is storing a boolean value, or adding a `debug_assert!(kernel_stack >= 1)`?

Same would go for the other `bool` to `u32` transitions.

Reply:
Thank you for your feedback! :+1: I added a explicit comment stating that the variable is storing a boolean value.